src/discovery.rs

```diff
@@ -387,27 +388,34 @@ impl Context for OidcDiscovery {
                             return;
                         }
 
-                       // For all keys, check if it is a key of alg RS256 and append it to the list of keys.
-                       let mut keys : Vec<jwt_simple::algorithms::RS256PublicKey> = Vec::new();
+                       // For all keys, check if it is a key of any of the supported types.
+                       let mut keys : Vec<SigningKey> = vec![];
                        for key in jwks_response.keys {
-                            if key.kty == "RSA" && key.alg == "RS256" {
-                                // If the key id is present, set the state to ready and return.
-                                // Extract public key components
-                                let public_key_comp_n = &key.n;
-                                let public_key_comp_e = &key.e;
-
-                                // Decode and parse the public key
-                                let n_dec = base64engine_urlsafe.decode(public_key_comp_n).unwrap();
-                                let e_dec = base64engine_urlsafe.decode(public_key_comp_e).unwrap();
-
-                                let public_key = jwt_simple::algorithms::RS256PublicKey::from_components(&n_dec, &e_dec)
-                                .unwrap();
-
-                                keys.push(public_key);
-                            }
-                            // Possibilty to add more algorithms here
-                            else {
-                                warn!("unsupported algorithm: {}", key.alg);
+                            match key {
+                                // Build the public key from the components and add it to the list of keys.
+                                JsonWebKey::RS256{kty, alg, n, e} => {
+
+                                    // Check if the key is of type RSA
+                                    if kty != "RSA" && alg != "RS256" {
+                                        warn!("key is not of type RSA or alg is not RS256, skipping");
+                                        continue;
+                                    }
+
+                                    // Extract public key components
+                                    let public_key_comp_n = &n;
+                                    let public_key_comp_e = &e;
+
+                                    // Decode and parse the public key
+                                    let n_dec = base64engine_urlsafe.decode(public_key_comp_n).unwrap();
+                                    let e_dec = base64engine_urlsafe.decode(public_key_comp_e).unwrap();
+
+                                    let public_key = SigningKey::RS256PublicKey(
+                                        jwt_simple::algorithms::RS256PublicKey::from_components(&n_dec, &e_dec)
+                                    .unwrap());
+
+                                    keys.push(public_key);
```

Comment:
I would move this out of the match statement, then the entire match statement could be a function. You could even `impl From<&JsonWebKey> for SigningKey`.

Reply:
Done :) @miwig